llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/DependencyGraph.h

```diff
@@ -159,8 +168,37 @@ class DependencyGraph {
   /// The DAG spans across all instructions in this interval.
   Interval<Instruction> DAGInterval;
 
+  std::unique_ptr<BatchAAResults> BatchAA;
+
+  enum class DependencyType {
+    RAW,   ///> Read After Write
+    WAW,   ///> Write After Write
+    RAR,   ///> Read After Read
```

Comment:
can we group RAR into None?

Reply:
We could, but None has a different meaning, that the instructions are not accessing memory.

Reply:
it looks like the current patch doesn't distinguish between the two. are we planning on distinguishing the two at some point? RAR isn't really a dependency right?